### PR TITLE
5 Year Reunion Submission

### PR DESCRIPTION
**Pull request by karenxiao**
Submission for the 5-Year Red Book

### CODE REVIEW

File: Karen.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+/**
+ * @author Karen Xiao
+ * @see 5th Anniversary Class Report
+ * @since 11/9/18
+ */
+
+public class Karen
+{
+    private Integer age;
```
Comment (kennyyu) [inline Karen.java]: "effective java" says you should use `final` for these :P

https://stackoverflow.com/questions/137868/using-the-final-modifier-whenever-applicable-in-java

File: Karen.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            yearInTheLifeOf(karen, freeTime.get(year));
+        }
+
+        for (int year = 2; year <= 5; year++)
+        {
+            karen.age++;
+            karen.job = "Engineer";
+            karen.company = "Mark43";
+
+            if (year = 4)
```
Comment (kennyyu) [inline Karen.java]: this is amazing!! 

shipit

File: Karen.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    {
+        this.age = age;
+        this.currentCity = currentCity;
+        this.job = job;
+        this.company = company;
+    }
+
+    private static void yearInTheLifeOf(Karen k, String hobby)
+    {
+        System.out.println(StringUtils.join(Arrays.asList(
```
Comment (dillen-roggensinger) [inline Karen.java]: You could put the `toString` method on this `Karen` class and then make your main method look like this:
```
public static void main(String[] args)
{
    Karen karen = new Karen(20, "Cambridge, MA", "Student", "Harvard");
    karen.growUp();
}

//notice instance method
public void growUp()
{
    aYearInTheLifeOf(freeTime.get(0));
    
    for (int year = 1; year < 2; year++)
    {
         ...
    }
}

public void aYearInTheLifeOf(String hobby)
{
    System.out.println(this.toString() + " | " + hobby);
}

public void toString()
{
   //join thing here
}
```

File: Karen.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+public class Karen
+{
+    private Integer age;
+    private String currentCity;
+    private String job;
+    private String company;
+    private static List<String> freeTime = Arrays.asList("Travel", "Dance", "Eat Hot Pot", "Paint", "Run", "Climb");
+
+    private Karen(Integer age, String currentCity, String job, String company)
```
Comment (dillen-roggensinger) [inline Karen.java]: Don't use `Integer` unless you ever expect your age to be `null`

File: Karen.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+ * @since 11/9/18
+ */
+
+public class Karen
+{
+    private Integer age;
+    private String currentCity;
+    private String job;
+    private String company;
+    private static List<String> freeTime = Arrays.asList("Travel", "Dance", "Eat Hot Pot", "Paint", "Run", "Climb");
```
Comment (dillen-roggensinger) [inline Karen.java]: should be a final constant. I'd lift it to the top of the class and capitalize it: `private static final List<String> FREE_TIME`

File: Karen.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+    public static void main(String[] args)
+    {
+        Karen karen = new Karen(20, "Cambridge, MA", "Student", "Harvard");
+
+        yearInTheLifeOf(karen, freeTime.get(0));
+
+        for (int year = 1; year < 2; year++)
+        {
+            karen.age++;
```
Comment (dillen-roggensinger) [inline Karen.java]: you could make these all methods like `getOlder(int years)` or `switchJobs(String job)` or `moveCities(String city)` etc

**Review by dillen-roggensinger — APPROVED** [review]
This is amazing and a really clever idea. You could make the code super enterprise-y or or more object oriented if you'd like but it's great as is 🥇

**Pull request merged**